### ISSUE TITLE
podAntiAffinityLabelSelector and podAntiAffinityLabelSelector a bit broken.

### ISSUE THREAD

**Issue by kramvan1** [istio/istio#13892]
**Bug description**

I think this pull request only did half the job: https://github.com/istio/istio/pull/12798/files
The `gatewaypodAntiAffinityRequiredDuringScheduling` and `podAntiAffinityLabelSelector ` are  still broken.

AND I think the wrong default are being used 
`podAntiAffinityLabelSelector: {}` but should be `podAntiAffinityLabelSelector: []` as this is used as a list.    This works for the empty case since the list/map are empty.
But this warning shows up with helm trying to use it:  
```
`Warning: Merging destination map for chart 'istio'. Cannot overwrite table item 'podAntiAffinityTermLabelSelector', with non table value: map[]`  
```

**Expected behavior**

**Steps to reproduce the bug**

values.yaml
```
pilot:
  podAntiAffinityTermLabelSelector:
    - key: app
      operator: in
      values: pilot
      topologyKey: "kubernetes.io/hostname"
```
**Version (include the output of `istioctl version --remote` and `kubectl version`)**
1.1.5 .  and Master

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

**Affected product area (please put an X in all that apply)**

[x] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by kramvan1** [comment]
cc @linsun

**Comment by linsun** [comment]
good feedback thank you @kramvan1 !

@morvencao could you try fix this in 1.1, assuming it is small fix?

**Comment by morvencao** [comment]
Please check https://github.com/istio/istio/pull/13905 for details fix.

Local test with values for pilot:
```
podAntiAffinityLabelSelector:
  - key: pilot
    operator: In
    values: S1,S2
    topologyKey: "kubernetes.io/hostname"
podAntiAffinityTermLabelSelector:
  - key: app
    operator: In
    values: pilot
    topologyKey: "kubernetes.io/hostname"
```

`helm template...` gets the following result:
```
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: pilot
                operator: In
                values:
                - S1
                - S2
            topologyKey: kubernetes.io/hostname
          preferredDuringSchedulingIgnoredDuringExecution:
          - podAffinityTerm:
              labelSelector:
                matchExpressions:
                - key: app
                  operator: In
                  values:
                  - pilot
              topologyKey: kubernetes.io/hostname
```

**Comment by rvennam** [comment]
@kramvan1 Can you please verify the fix?

**Comment by howardjohn** [comment]
Fixed by #13905
